app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -0,0 +1,619 @@
+  private static final long TIME_BETWEEN_SCROLL = 2 * DateUtils.SECOND_IN_MILLIS;
+  private static final String TAG = AppViewPresenter.class.getSimpleName();
+
+  private final PermissionManager permissionManager;
+  private final PermissionService permissionService;
+  private AppViewView view;
+  private AccountNavigator accountNavigator;
+  private AppViewAnalytics appViewAnalytics;
+  private AppViewNavigator appViewNavigator;
+  private AppViewManager appViewManager;
+  private AptoideAccountManager accountManager;
+  private Scheduler viewScheduler;
+  private CrashReport crashReport;
+  private long appId;
+  private String packageName;
+
+  public AppViewPresenter(AppViewView view, AccountNavigator accountNavigator,
+      AppViewAnalytics appViewAnalytics, AppViewNavigator appViewNavigator,
+      AppViewManager appViewManager, AptoideAccountManager accountManager, Scheduler viewScheduler,
+      CrashReport crashReport, long appId, String packageName, PermissionManager permissionManager,
+      PermissionService permissionService) {
+    this.view = view;
+    this.accountNavigator = accountNavigator;
+    this.appViewAnalytics = appViewAnalytics;
+    this.appViewNavigator = appViewNavigator;
+    this.appViewManager = appViewManager;
+    this.accountManager = accountManager;
+    this.viewScheduler = viewScheduler;
+    this.crashReport = crashReport;
+    this.appId = appId;
+    this.packageName = packageName;
+    this.permissionManager = permissionManager;
+    this.permissionService = permissionService;
+  }
+
+  @Override public void present() {
+    handleFirstLoad();
+    handleReviewAutoScroll();
+    handleClickOnScreenshot();
+    handleClickOnVideo();
+    handleClickOnDescriptionReadMore();
+    handleClickOnDeveloperWebsite();
+    handleClickOnDeveloperEmail();
+    handleClickOnDeveloperPrivacy();
+    handleClickOnDeveloperPermissions();
+    handleClickOnStoreLayout();
+    handleClickOnFollowStore();
+    handleClickOnOtherVersions();
+    handleClickOnTrustedBadge();
+    handleClickOnRateApp();
+    handleClickReadComments();
+    handleClickFlags();
+    handleClickLoginSnack();
+    handleClickOnSimilarApps();
+    handleClickOnToolbar();
+    handleDefaultShare();
+    handleRecommendsShare();
+    handleClickOnRetry();
+
+    handleInstallButtonClick();
+    pauseDownload();
+    resumeDownload();
+    cancelDownload();
+    loadDownloadApp();
+  }
+
+  private void handleFirstLoad() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .doOnNext(__ -> view.showLoading())
+        .flatMap(__ -> loadApp())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private void handleReviewAutoScroll() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.scrollReviewsResponse())
```

Comment:
scrollReviewsResponse name is poor. If the reviews animation setup depend on the first getApp, this should probably go to the chain where the reviews are being shown? Is It possible?